torchinductor/dependencies.py

```diff
@@ -14,23 +20,25 @@
 
 log = logging.getLogger(__name__)
 
+Dep = Union["MemoryDep", "StarDep"]
```

Comment:
What do folks think?

MemoryDep and StarDep seem to be meaningfully different (i.e. StarDep does not have the notion of an index or a size), and there's a bunch of places that currently are annotated as "MemoryDep" that I suspect can actually hold StarDep too.

Reply:
It makes sense, but I would prefer to explicitly write `Dep` as a base class.

Reply:
+1 to add a base class